src/main/java/org/kuali/kfs/kim/inquiry/PersonInquirableImpl.java

```diff
@@ -0,0 +1,192 @@
+ */
+package org.kuali.kfs.kim.inquiry;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.kuali.kfs.kim.impl.common.attribute.KimAttributeDataBo;
+import org.kuali.kfs.kim.impl.identity.PersonImpl;
+import org.kuali.kfs.kim.impl.type.KimTypeAttributeBo;
+import org.kuali.kfs.kns.datadictionary.InquirySectionDefinition;
+import org.kuali.kfs.kns.inquiry.InquiryRestrictions;
+import org.kuali.kfs.kns.inquiry.KualiInquirableImpl;
+import org.kuali.kfs.kns.lookup.HtmlData;
+import org.kuali.kfs.datadictionary.legacy.DataDictionaryService;
+import org.kuali.kfs.kns.service.KNSServiceLocator;
+import org.kuali.kfs.kns.web.ui.Field;
+import org.kuali.kfs.kns.web.ui.Row;
+import org.kuali.kfs.kns.web.ui.Section;
+import org.kuali.kfs.kns.web.ui.SectionBridge;
+import org.kuali.kfs.krad.util.GlobalVariables;
+import org.kuali.rice.kim.api.identity.Person;
+import org.kuali.rice.kim.api.identity.PersonService;
+import org.kuali.rice.kim.api.services.KimApiServiceLocator;
+import org.kuali.rice.krad.bo.BusinessObject;
+
+import java.util.AbstractMap;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Comparator;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class PersonInquirableImpl extends KualiInquirableImpl {
+
+    private static final Logger LOG = LogManager.getLogger();
+    private static final Map<String, String> SUB_SECTION_LABELS = Stream.of(
+            new AbstractMap.SimpleEntry<>("rolesSection", "Role Qualifier"),
+            new AbstractMap.SimpleEntry<>("delegationsSection", "Delegation Member Qualifier"))
+            .collect(Collectors.toMap(AbstractMap.SimpleEntry::getKey, AbstractMap.SimpleEntry::getValue));
+
+    PersonService personService;
+
+    @Override
+    public BusinessObject getBusinessObject(Map fieldValues) {
+        Person person = getPersonService().getPerson(fieldValues.get("principalId").toString());
+        // KFSPTS-19308 CU customization to allow inquiry by prinicpalName
+        if (person == null && fieldValues.containsKey("principalName")) {
+            String principalName = fieldValues.get("principalName").toString();
+            if (StringUtils.isNotBlank(principalName)) {
+                person = getPersonService().getPersonByPrincipalName(principalName);
+            }
+        }
+        
+        if (person instanceof PersonImpl) {
+            ((PersonImpl) person).populateMembers();
+        }
+        return person;
+
+    }
+
+    @Override
+    public HtmlData getInquiryUrl(BusinessObject businessObject, String attributeName, boolean forceInquiry) {
+        List<String> primaryKeys = new ArrayList<>();
+        primaryKeys.add("principalId");
+        //KFSPTS-19308 CU customization to allow inquiry by prinicpalName
+        primaryKeys.add("principalName");

```

Comment:
Were you specifically encountering inquiry problems if the principalName was not being included in this manner? I'm reluctant about treating non-PK fields as primary keys, given that principalId is the actual primary key (despite principalName also being unique to each user).

Also, there's a small typo in the explanatory comment.